File: python/Leapyear04.py
import calendar

year = int(input("Enter the year to check"))

def printLeapYear01(year):
    if year % 400 == 0 or (year % 100 != 0 and year % 4 == 0):
        return "Leap Year"
    else:
        return "Not Leap Year"
    
def printLeapYear02(year):
    return "Leap Year" if (year % 400 == 0 or (year % 100 != 0 and year % 4 == 0)) else "Not Leap Year"    


def printLeapYear03(year):
    return "Leap Year" if (year % 400 == 0 or (year % 100 != 0 and year % 4 == 0)) else "Not Leap Year"    

def printLeapYear04(year):
    return "Leap Year" if calendar.isleap(year) else "Not Leap Year"

print(printLeapYear01(year))    
print(printLeapYear02(year))
print(printLeapYear03(year))
print(printLeapYear04(year))    

File: java/basics/Leapyear04.java
package basics;

import java.time.Year;
import java.util.Scanner;

public class Leapyear04 {

	public static void main(String[] args) {
		System.out.println("Enter he year to check");
		Scanner scn = new Scanner(System.in);
		int year = scn.nextInt();
		scn.close();
		System.out.println(printLeapYear01(year));
		System.out.println(printLeapYear02(year));
		System.out.println(printLeapYear03(year));
		
	}

	private static String printLeapYear01(int year) {
		if (year % 400 == 0 || (year % 100 != 0 && year % 4 == 0)) {
			return "Leap Year";
		} else {
			return "Not Leap Year";
		}
	}
	
	private static String printLeapYear02(int year) {
		 return (year % 400 == 0 || (year % 100 != 0 && year % 4 == 0))?"Leap Year":"Not Leap Year";		
	}
	
	private static String printLeapYear03(int year) {
		Year yearOb = Year.of(year);
		return yearOb.isLeap() ? "Leap year" : "Not Leap Year";
	}

}
